### PR TITLE
feat: add gloo support with numpy

### PR DESCRIPTION
**Pull request by zw0610**
**What this PR does / why we need it**:

Add Gloo support

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @terrytangyuan 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: ftlib/commlib/gloo/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.8.1)
+
+project(gloo_lib LANGUAGES CXX)
+
+find_package(pybind11)
+pybind11_add_module(gloo_lib src/gloo_lib.cxx)
+
+# we are not building with nccl
```
Comment (terrytangyuan) [inline ftlib/commlib/gloo/CMakeLists.txt]: Can these comments be removed?

File: ftlib/commlib/gloo/src/gloo_lib.cxx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <chrono>
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "gloo/transport/tcp/device.h"
```
Comment (terrytangyuan) [inline ftlib/commlib/gloo/src/gloo_lib.cxx]: Where did Gloo get installed? e.g. if users want to use this, how should they install FTLib to be able to use Gloo implementation?

File: ftlib/commlib/gloo/CMakeLists.txt (hunk 1 above)
Comment (zw0610) [reply ftlib/commlib/gloo/CMakeLists.txt]: sure

File: ftlib/commlib/gloo/src/gloo_lib.cxx (hunk 2 above)
Comment (zw0610) [reply ftlib/commlib/gloo/src/gloo_lib.cxx]: I'll add a dockerfile to this pr.

**Pull request merged**